presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -119,6 +120,15 @@ public DbResourceGroupConfigurationManager(ClusterMemoryPoolManager memoryPoolMa
         if (lastRefresh.get() == 0) {
```

Comment:
I'd just change this to a `checkArgument(lastRefresh.get() != 0)` or remove it altogether. It's not very useful. If `lastRefresh` is 0 the new logic would throw anyways. You can change the error message to print something more reasonable for this special case.

Reply:
It means resource group has never successfully being fetched -- this means something is severely wrong but it could happen right?

Reply:
Well at least currently you are throwing the same exception in both cases, thus nothing distinguishes the "severity". Removing this and and modify the error message to handle `lastRefresh = 0` has the same effect. If you actually believe it should be more severe, handle it accordingly. 😛

Reply:
@rongrong : Ah I see. From the code perspective is about whether "lastReferesh == 0" means something special case :). 

I don't have strong opinion here, although my personal coding style will handle it separately :). Anyhow, I will put it as a special case inside "System.nanoTime() - lastRefresh.get() > maxRefreshInterval.toMillis() * 1_000_000"